helix-gateway/src/main/java/org/apache/helix/gateway/grpcservice/HelixGatewayServiceGrpcService.java

```diff
@@ -73,17 +75,20 @@ public void onCompleted() {
   /**
    * Send state transition message to the instance.
    * The instance must already have established a connection to the gateway service.
-   * @param instanceName
-   * @return
+   * @param instanceName the instance name to send the message to
+   * @param messageType the type of the message
+   * @param message the message to convert to the transition message
    */
   @Override
-  public boolean sendStateTransitionMessage(String instanceName) {
+  public void sendStateTransitionMessage(String instanceName, MessageType messageType,
```

Comment:
Maybe we do not need to pass the MessageType here. we could read the from and f=to state In translation util. Message type also confuses with org.apache.helix.model.Message.MessageType.